src/node_crypto.cc

```diff
@@ -212,6 +212,111 @@ static int NoPasswordCallback(char* buf, int size, int rwflag, void* u) {
 }
 
 
+// namespace node::crypto::error
+namespace error {
+Local<Object> Decorate(Environment* env, Local<Object> obj,
+                             unsigned long err) {  // NOLINT(runtime/int)
+  if (err == 0) return obj;  // No decoration possible.
+
+  const char* ls = ERR_lib_error_string(err);
+  const char* fs = ERR_func_error_string(err);
+  const char* rs = ERR_reason_error_string(err);
+
+  Isolate* isolate = env->isolate();
+  Local<Context> context = isolate->GetCurrentContext();
+
+  if (ls != nullptr)
+    obj->Set(context, env->library_string(),
+        OneByteString(isolate, ls)).FromJust();
+  if (fs != nullptr)
+    obj->Set(context, env->function_string(),
+        OneByteString(isolate, fs)).FromJust();
+  if (rs != nullptr) {
+    obj->Set(context, env->reason_string(),
+        OneByteString(isolate, rs)).FromJust();
```

Comment:
> (And I’m mostly concerned about the new .FromJust()s being added here, not so much this particular bit here.)

Is this the code you are commenting about @addaleax ? `FromJust()` is called on pretty much every `Set()` in node_crypto.cc, I  thought it is what we are supposed to do. Is it all wrong, or just here? What do you want changed? The `FromJust()` just removed? Something else? I'll make the changes but I need more information.

Btw, this code is an exact copy of code you read and LGTMed in https://github.com/nodejs/node/pull/25093#pullrequestreview-187419311 :-(.

Reply:
> > (And I’m mostly concerned about the new .FromJust()s being added here, not so much this particular bit here.)
> 
> Is this the code you are commenting about @addaleax ? `FromJust()` is called on pretty much every `Set()` in node_crypto.cc, I thought it is what we are supposed to do. Is it all wrong, or just here? What do you want changed? The `FromJust()` just removed? Something else? I'll make the changes but I need more information.

I think the other comment I just posted might help with explaining the justification a bit more… what I’d think would be best is to return an empty `MaybeLocal<>`, without futher calls into V8, if `->Set()` returns an empty `Maybe<>` (i.e. `obj->Set(...).IsNothing()` returns true).

> 
> Btw, this code is an exact copy of code you read and LGTMed in [#25093 (review)](https://github.com/nodejs/node/pull/25093#pullrequestreview-187419311) :-(.

I’m sorry about that, I should probably have pointed that out there – although you’re right in that currently a lot of code follows this style…

Reply:
ok, see fixup